forms/HtmlEditorField.php

```diff
@@ -578,7 +578,7 @@ public function getanchors() {
 
 			// Similar to the regex found in HtmlEditorField.js / getAnchors method.
 			if (preg_match_all("/\s(name|id)=\"([^\"]+?)\"|\s(name|id)='([^']+?)'/im", $page->Content, $matches)) {
```

Comment:
I thought the same thing; but cleaning up the regex to something like:

`/\s(name|id)\s*=\s*["']?([^'"]+?)['"]?\s*/` (ie: one unified regex) means that an attribute like so wouldn't be matched properly: `<a name="test's">` as the `'` would cause the regex to return `test` instead of `test's`